src/script/conversation/ConversationStateHandler.ts

```diff
@@ -89,14 +89,13 @@ export class ConversationStateHandler extends AbstractConversationEventHandler {
 
         if (accessModes && accessRole) {
           try {
-            await this.conversationService.putConversationAccess(conversationEntity.id, accessModes, accessRole);
-
-            conversationEntity.accessState(accessState);
-
             if (accessState === ACCESS_STATE.TEAM.TEAM_ONLY || accessState === ACCESS_STATE.TEAM.SERVICES) {
               conversationEntity.accessCode(undefined);
-              // this.revokeAccessCode(conversationEntity);
+              await this.revokeAccessCode(conversationEntity);
```

Comment:
We need to revoke the access code before actually changing the conversation state (else the backend will reply with a 403)